### PR TITLE
Feature/go dal

### PR DESCRIPTION
**Pull request by stgmsa**
# Description
implement data access layer using go for table `admin_api_audit_log`

# Impacts
* basic CRUDs accessing `admin_api_audit_log`
* URL queries for List API
* JSON payload query for Search API 
* edge cases and error handlings

# NEW Package(s) required
* validator v10

# Delete branch after merge
YES

# Checklist
- [yes] Document the feature
- [yes] Add unit tests
- [no] Add acceptance tests (TestLink)

# NEWS file entries
## New Features
* Data access layer implemented using go

### CODE REVIEW

File: go/caddy/admin-api-audit-log/models/models.go [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+	"errors"
+	"github.com/inverse-inc/packetfence/go/caddy/pfpki/sql"
+	"github.com/jinzhu/gorm"
+	"strings"
+	"time"
+)
+
+type AdminApiAuditLog struct {
+	ID        int64     `json:"id,omitempty" gorm:"primary_key"`
+	CreatedAt time.Time `json:"created_at,omitempty"`
```
Comment (jrouzierinverse) [inline go/caddy/admin-api-audit-log/models/models.go]: Change to a pointer to avoid it being displayed when not in fields.
```
CreatedAt *time.Time `json:"created_at,omitempty"`
```

File: go/caddy/api/dal_admin_api_audit_log_test.go [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+func TestList(t *testing.T) {
+
+	db := getGormDB(t)
+
+	log := admin_api_audit_log.AdminApiAuditLog{
+		Method: "POST",
+		Status: 200,
+	}
+	err := admin_api_audit_log.Add(db, &log)
+	fmt.Println(err)
```
Comment (jrouzierinverse) [inline go/caddy/api/dal_admin_api_audit_log_test.go]: Handle error here

File: go/caddy/api/dal_admin_api_audit_log_test.go [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+	w := httptest.NewRecorder()
+	handler(w, req)
+	res := w.Result()
+	defer res.Body.Close()
+	data, err := ioutil.ReadAll(res.Body)
+	if err != nil {
+		t.Fatalf("Error: %s", err.Error())
+	} else {
+		//Add additional checks
+		fmt.Printf("Data: %s\n", string(data))
```
Comment (jrouzierinverse) [inline go/caddy/api/dal_admin_api_audit_log_test.go]: Replace with 
```go
t.Logf("Data: %s\n", string(data))
```

**Review by jrouzierinverse — CHANGES_REQUESTED** [review]

File: go/caddy/admin-api-audit-log/models/models.go
Comment (jrouzierinverse) [inline go/caddy/admin-api-audit-log/models/models.go:1]: Rename this file and make a different file for each model.
go/caddy/admin-api-audit-log/models/models.go => go/dal/models

**Review by jrouzierinverse — CHANGES_REQUESTED** [review]

File: go/caddy/admin-api-audit-log/models/models.go
Comment (stgmsa) [reply go/caddy/admin-api-audit-log/models/models.go:1]: moved to go/caddy/dal/models

**Comment by jrouzierinverse** [comment]
Remove the DELETE and PATCH methods for the following tables.

admin_api_audit_log
auth_log
dns_audit_log
radacct_log
radius_audit_log

**Comment by stgmsa** [comment]
> 

removed & ready for re-review.
